src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -232,6 +232,12 @@ internal Binder GetSpeculativeBinder(int position, ExpressionSyntax expression,
 
             binder = new WithNullableContextBinder(SyntaxTree, position, binder);
 
+            // PROTOTYPE(semi-auto-props): We should traverse until we get a property accessor, similar to Binder_Expressions. We may want to share the code.
+            if (binder.ContainingMemberOrLambda is SourcePropertyAccessorSymbol { Property.IsIndexer: false } accessor)
+            {
+                binder = new SpeculativeFieldKeywordBinder(accessor, binder);
```

Comment:
>binder = new SpeculativeFieldKeywordBinder(accessor, binder); [](http://example.com/codeflow?start=16&length=61)

How certain are we that we intercepted all the code paths that can do speculation?

Reply:
Unfortunately I cannot find a "centralized" place where I could assert that. The assert you suggested in `MethodBodySemanticModel` can help, but still no guarantee unfortunately.